src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftDeployer.java

```diff
@@ -98,10 +98,11 @@ public boolean coreLogic(Launcher launcher, TaskListener listener,
         			}
         			
     				try {
-    					rc = this.getLatestReplicationController(dc, client, overrides, chatty ? listener : null);
+    					rc = Deployment.getLatestReplicationController(dc, getNamespace(overrides), client, chatty ? listener : null);
     					if (chatty)
     						listener.getLogger().println("\nOpenShiftDeployer returned rep ctrl " + rc);
     					if (rc != null) {
+    	    				AnnotationUtils.AnnotateResource(client, listener, chatty, env, rc); 
```

Comment:
@gabemontero unless this annotation call will always ensure the RC is properly annotated upon being triggered by a deployment step?